#!/bin/bash
#swarm -f 1_CombAll_novoalign_uniquebam_Find_Peaks.sh --verbose 1 -g 50 -t 32 -b 20 --time 6:00:00 --job-name clippeaks --module samtools,R,bedtools
  
 ###################################
 ### Identify peaks
 
samtools view -H /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam > /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.header.txt

samtools sort /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam | samtools view | awk -F'\t' 'function abs(x){return ((x < 0.0) ? -x : x)} {if (abs($9) < 1000000) print $0}' | cat /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.header.txt - | samtools view -Sb > /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.bam

bedtools bamtobed -i /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.bam > /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.bed

#bedtools merge -c 6 -o count,distinct -d -1 -i /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.bed> /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks_noS.txt
#bedtools merge -c 1,6 -o count,distinct -d -1 -i /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.bed> /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000_noS.peaks2.txt
bedtools merge -c 6 -o count,distinct -d -1 -i /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.bed -s > /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks.txt
#bedtools merge -c 1,6 -o count,distinct -d -1 -i /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.bed -s > /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks2.txt

Rscript /data/RBL_NCI/Wolin/Phil/mESC_clip/scripts/SelectPeaks.R /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks.txt /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/
 
 
 ###################################
 ### Peak counts
 bedtools coverage -f 0.0001 -counts -a /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks.bed -b /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam -s > /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.peakscount.txt
 
Rscript /data/RBL_NCI/Wolin/Phil/mESC_clip/scripts/UntNts_CIGAR_V6b_count.R /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks.bed /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/peaks/ samtools /data/RBL_NCI/Wolin/Phil/mESC_nascent/reference/fasta/mm10/fdb_igenomes_Mus_musculus_UCSC_mm10_Sequence_WholeGenomeFasta_genome.fa /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam 1 1 0 0 NA .PeakcountsUniq

############################################
### Control Samples

#Rscript /data/RBL_NCI/Wolin/Phil/mESC_clip/scripts/UntNts_CIGAR_V6b_count.R /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks.bed /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique samtools /data/RBL_NCI/Wolin/Phil/mESC_nascent/reference/fasta/mm10/fdb_igenomes_Mus_musculus_UCSC_mm10_Sequence_WholeGenomeFasta_genome.fa /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/clip_Control_novoalign_random.bam 1 1 0 0 NA .Peakcounts


#bedtools coverage -f 0.0001 -d -a /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/uniqueclip_Control_novoalign_random.bam -b /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks.bed -s > /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks2xxxx.txt

#bedtools coverage -f 0.0001 -counts -a /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/uniqueclip_Control_novoalign_random.bam -b /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks.bed -s > /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks2xxxx.txt

#bedtools coverage -f 0.0001 -counts -a /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/Ro_Clip_iCountcutadpt_all.unique.NH.ddup.s.bam.tlen1000000.peaks.bed -b /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/uniqueclip_Control_novoalign_random.bam -s > /data/RBL_NCI/Wolin/Phil/mESC_clip/Novo_UMI_split/bam_icountprocess/mm10_Gencode_rRNA/iCount_all/sep_MM_Uniq/unique/clip_Control_novoalign_random.peakscount.txt
